### PR TITLE
Add docs for annotations

### PR DESCRIPTION
**Pull request by JureZmrzlikar**

### CODE REVIEW

**Comment by JureZmrzlikar** [comment]
@mzganec @mstajdohar 
After today's discussion, please review this implementation again - does it suit the requirements you have expressed?

File: docs/tutorial-create.rst [hunk 1]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
+
+
+Annotation Fields
+-----------------
 
-We can annotate the sample object using a similar process with a 'sample'
-descriptor schema:
+You might be wondering why the example above requires ``general.species`` string
+instead of e.g. just ``species``. The answer to this are ``AnnotationFields``\ s.
+These are predefined set of names that are available to annotate samples. Let's
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:97]: Annotation field represents more than just a name. Maybe predefined set of attribute/characteristics/terms?

File: docs/tutorial-create.rst [hunk 2]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
+They connect the field with the actual value.
+
+.. literalinclude:: files/tutorial-create.py
+   :lines: 48-55
 
-We can also define descriptors and descriptor schema directly when calling
-``res.run`` function. This is described in the section about the ``run()``
-method below.
+
+As a shortcut, you can get all the ``AnnotationValue``\ s for a given sample by
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:131]: For consistency add punctuation mark at the end. In other similar cases there is colon at the end.

File: docs/tutorial-create.rst [hunk 3]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
+.. literalinclude:: files/tutorial-create.py
+   :lines: 57
+
+
+Helper methods
+--------------
+
+Sometimes it is convenient to represent the annotations with the dictionary,
+where keys are field names and values are annotation values. You can get all
+the annotation for a given sample in this format by calling
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:142]: Missing punctuation mark.

File: docs/tutorial-create.rst [hunk 4]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
+
+.. literalinclude:: files/tutorial-create.py
+   :lines: 58
+
+Multiple annotations stored in the dictionary can be assigned to sample by
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:147]: Missing punctuation mark.

File: docs/tutorial-create.rst [hunk 5]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
-`Resolwe Bioinformatics documentation`_.
+Annotate Samples
+================
 
-.. _Resolwe Bioinformatics documentation: http://resolwe-bio.readthedocs.io
-.. _Descriptor schemas: https://resolwe-bio.readthedocs.io/en/latest/descriptor.html
-.. _accompanying examples: https://github.com/genialis/resolwe-bio/tree/master/resolwe_bio/descriptors
+The next thing to do after uploading some data is to annotate samples this data
+belongs to. This can be done by assigning a value to a predefined field on a
+given sample. See example below.
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:82]:
```suggestion
given sample. See the example below.
```
Missing the.

File: docs/tutorial-create.rst [hunk 6]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
 
-.. _sample descriptor schema: https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/descriptors/sample.yml
+If you wish to examine what fields are available, use a query
+
+.. literalinclude:: files/tutorial-create.py
+   :lines: 44-46
+
+
+You may be wondering whether you an create your own fields / groups. The answer
+is no. Time has proven that keeping things organized requires usage
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:115]:
```suggestion
is no. Time has proven that keeping things organized requires the usage
```

File: docs/tutorial-create.rst [hunk 7]
```diff
@@ -74,50 +74,86 @@ applied on the server.
...
+   Each field is uniquely defined by the combination of ``name`` and ``group``.
 
-.. _sample descriptor schema: https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/descriptors/sample.yml
+If you wish to examine what fields are available, use a query
+
+.. literalinclude:: files/tutorial-create.py
+   :lines: 44-46
+
+
+You may be wondering whether you an create your own fields / groups. The answer
```
Comment (AnzeLovse) [inline docs/tutorial-create.rst:114]:
```suggestion
You may be wondering whether you can create your own fields / groups. The answer
```

File: docs/files/tutorial-create.py [hunk 8]
```diff
@@ -30,39 +30,39 @@
...
+reads.sample.set_annotation("general.descriptor", None)
 
-reads.sample.save()
-
-
-
-
-
-
+65
```
Comment (AnzeLovse) [inline docs/files/tutorial-create.py:65]: Why is this needed?

**Review by AnzeLovse — COMMENTED** [review]
The guide explains the essential functionality well and I find the it helpful to familiarize with the concept. I only added a few grammatical nitpicks.

File: docs/tutorial-create.rst (hunk 1 above)
Comment (JureZmrzlikar) [reply docs/tutorial-create.rst:97]: Valid point, thanks. It tried to generalize the description without making it too complex.

File: docs/files/tutorial-create.py (hunk 8 above)
Comment (JureZmrzlikar) [reply docs/files/tutorial-create.py:65]: Not needed, this is an artefact after I was updating the code.

**Comment by JureZmrzlikar** [comment]
Thanks for the review @AnzeLovse . I modified the code with your suggestions.

**Review by AnzeLovse — APPROVED** [review]

**Review by mzganec — APPROVED** [review]
Looks good!
